### ISSUE TITLE
2019_R2 broken with some models

### ISSUE THREAD

**Issue by r0l1** [opencv/dldt#257]
Hi,
I am working with some tensorflow object detection models (SSD MobileNetv2) on a Myriad X and two of three models don't work with the latest 2019_R2 release. However everything is fine with 2019_R1.1.

The inference engine just hangs for ever. I am using it together with the OpenCV DNN backend. I'll post some code and models soon. But maybe this is already a known issue, so please let me know :)

Edit: The Myriad firmware seams to have changed since R2.
- R1: `MvNCAPI-ma2480.mvcmd`
- R2: `MvNCAPI-ma2x8x.mvcmd`

**Comment by r0l1** [comment]
@skaldesh Adding you to the conversation.

**Comment by shubha-ramani** [comment]
Dear @r0l1 

It's definitely interesting that it used to work fine in R1.1 and suddenly doesn't in 2019R2. Please definitely provide more details. Since it's MYRIAD specific I don't think using Inference Engine API versus OpenCV DNN backend would make a difference.

Please post some simple code which demonstrates the issue and yes, your models and your mo commands as well.

Thanks for your patience,

Shubha

**Comment by r0l1** [comment]
Hi @shubha-ramani.
thank you for the fast response!
[Here is a demo application](http://cloud.desertbit.com/s/dtPi95KFwpQ9RWw) with an integrated model. Just tested it with R1.1 and R2 and I can confirm the problem.

Hope this helps to solve this issue.
Thanks!
Roland

**Comment by shubha-ramani** [comment]
Dear Roland @r0l1 
Thank you for your fast response too ! I definitely will take a look today. Please look for a status update here on this github ticket.

Shubha

**Comment by shubha-ramani** [comment]
Dear @r0l1 
I looked at your demo application and realized that unfortunately, this is the wrong github forum. You are not using Model Optimizer, nor are you using an Inference Engine backend. You are using OpenCV. Please post your issue at this forum:
[https://github.com/opencv/opencv/issues](https://github.com/opencv/opencv/issues). They will be better equipped to help you.

Sorry for the inconvenience,

Shubha

**Comment by r0l1** [comment]
Dear @shubha-ramani,
the OpenCV Inference Engine Backend is used:

```
net.setPreferableBackend(cv::dnn::DNN_BACKEND_INFERENCE_ENGINE);
net.setPreferableTarget(cv::dnn::DNN_TARGET_MYRIAD);
```

The problem only occurs with the movidius chip and only with the latest R2 release. I didn't test the model optimizer yet. The problem might be caused during the model preparation by the OpenCV backend.

I'll leave this issue open until I tested a converted model by the model optimizer. Thank you for your instant support.
Roland

**Comment by shubha-ramani** [comment]
Dear Roland @r0l1 
Thank you for understanding. What I meant was  Inference Engine Core API. Of course you're using IE backend via OpenCV. I misspoke with my terminology earlier.

I actually tried to get your project to run, despite that we normally don't non-IE-Core-API and non-Model Optimizer projects on this github. While I got past the "findPackage OpenCV" I kept getting IE linking errors.   I spent a couple of hours trying to fix them to no avail. I definitely set InferenceEngine_DIR, OpenCV_DIR etc in my terminal and even with your provided CMakeLists.txt file.

The issue is that I have hundreds of customers to support (not just you). So the easier you make it on me, the more likely it will be that I can reproduce your bug.

It would be ideal if you can either re-write your *.cpp to use IE Core API (this should be easy to do) so that I can just drop it in the samples folder and compile. Or make your OpenCV project self-contained so that it just works, tell me where to put it, etc...

Looking forward to hearing more,

Shubha

**Comment by r0l1** [comment]
Dear @shubha-ramani 
thank you for investing your time into this issue.

I will convert the models with the model optimizer this evening and try to reproduce the bug with the  Inference Engine Core API. Once ready, I'll let you know.

For the OpenCV sample, I could upload two docker containers as build environment for you. One with R1.1 and one with R2. Initially I also had problems to set everything up. The containers would simplify the build process. Is this an option for you?

Kind Regards
Roland

**Comment by shubha-ramani** [comment]
Dear @r0l1 
I prefer IE Core API as opposed to containers. When I file a bug, the engineering team prefers as few layers as possible. Which means no docker containers.

Shubha

**Comment by r0l1** [comment]
Dear @shubha-ramani this issue is fixed in OpenCV 4.1.2 together with OpenVino 2019R3. Thank you for your support! Have a great day.

Kind Regards
Roland
